ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/ConfiguredKsqlPlan.java

```diff
@@ -18,28 +18,29 @@
 import io.confluent.ksql.config.SessionConfig;
 import io.confluent.ksql.engine.KsqlPlan;
 import java.util.Objects;
+import java.util.Optional;
 
 public final class ConfiguredKsqlPlan {
 
-  private final KsqlPlan plan;
+  private final Optional<KsqlPlan> plan;
```

Comment:
Wouldn't it be better to have a `Optional<ConfiguredKsqlPlan>` that is passed to the KsqlEngine and SandboxedExecutionContext instead? A configured plan with no plan inside does not seem right.